### DISCUSSION TITLE
How to use parallel time series in GluonTS?

### DISCUSSION THREAD

**Discussion by jovidsilva** [awslabs/gluonts#1772] [Q&A]
Hi I wanted to use a single global model and train it on parallel time series. The model will then make predictions for each of the time series. 

1. Which models in GluonTS can I use?
2. Can I use LSNet, DeepAR (Non Vectored) and SimpleFeedForward Neural Networks?

**Comment by jaheba** [comment]
Hi,

I might misunderstand the question, but I think that's essentially the use case GluonTS was build around.

The [SageMaker DeepAR](https://docs.aws.amazon.com/sagemaker/latest/dg/deepar.html#deepar-inputoutput) is probably the best description of the data input format and what is supported. The main differences between GluonTS and SageMaker DeepAR is that we (currently) only support jsonlines as input and feature fields have different names.

All deep learning models in GluonTS should support this use case.

**Reply by jovidsilva** [reply]
Thanks, for the reply.

I wanted to take an input of multiple parallel time series with the same n time steps as below:

![image](https://user-images.githubusercontent.com/70814169/142394184-bfec9e46-6989-47a0-856e-47800b84ce78.png)

Then make predictions as follows (Highlighted in yellow)

![image](https://user-images.githubusercontent.com/70814169/142393970-7bb9c17b-3230-4034-8fe8-fc1e2f7bb31f.png)

**Comment by jaheba** [comment]
Right, so then the input data should look like:

```json
{"item_id": "time series 2", "start": "01/01/82", "target": [26, 28, 29, 28]}
{"item_id": "time series 2", "start": "01/01/82", "target": [26, 27, 26, 28]}
...
```

I would recommend going through the linked documentation, and look at this page for how to do inference: https://docs.aws.amazon.com/sagemaker/latest/dg/deepar-in-formats.html

**Comment by jovidsilva** [comment]
I understand how to represent the input data as an iterable that has a
“target” and a “start” field. But now how do I pass each of them to single
Estimator? Or do you suggest I set the model parameters and loop through
each series and make predictions. I want to pass all of the input time
series to a single estimator, is this possible?

**Comment by jaheba** [comment]
To quote our `README`:

```py
from gluonts.dataset.common import ListDataset
training_data = ListDataset(
    [{"start": df.index[0], "target": df.value[:"2015-04-05 00:00:00"]}],
    freq = "5min"
)

from gluonts.model.deepar import DeepAREstimator
from gluonts.mx.trainer import Trainer

estimator = DeepAREstimator(freq="5min", prediction_length=12, trainer=Trainer(epochs=10))
predictor = estimator.train(training_data=training_data)
```

Alternatively, there also is `FileDataset` for loading data from disk.

All which is required is to assemble the data into a `DataSet`, and pass it to the `.train` method of the estimator.
